app/lib/actions/middleware/recurring_logic.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Middleware
+    class RecurringLogic < ::Dynflow::Middleware
+
+      def delay(delay_options, *args)
```

Comment:
Some high-level comment describing the purpose of this middleware might help further generations to understand the intention